src/common/file_utils.cpp

```diff
@@ -4,15 +4,64 @@
 #include "common/string_utils.h"
 #include "glob/glob.hpp"
 
+#ifdef _WIN32
+#include <fileapi.h>
+#include <windows.h>
+#endif
+
 namespace kuzu {
 namespace common {
 
+FileInfo::~FileInfo() {
+#ifdef _WIN32
+    if (handle != nullptr) {
+        CloseHandle((HANDLE)handle);
+    }
+#else
+    if (fd != -1) {
+        close(fd);
+    }
+#endif
+}
+
+int64_t FileInfo::getFileSize() {
+#ifdef _WIN32
+    return GetFileSize((HANDLE)handle, nullptr);
+#else
+    struct stat s;
+    if (fstat(fd, &s) == -1) {
+        return -1;
+    }
+    return s.st_size;
+#endif
+}
+
 std::unique_ptr<FileInfo> FileUtils::openFile(const std::string& path, int flags) {
+#if defined(_WIN32)
+    // Not providing GENERIC_READ seems to cause problems.
+    int dwDesiredAccess = GENERIC_READ;
```

Comment:
Our convension is to use auto instead of the actual type

Reply:
That doesn't seem to be mentioned in the [style guide](https://github.com/kuzudb/kuzu/wiki/GraphflowDB-Coding-Style). Is the style guide out of date?